test/Npgsql.Tests/Types/ArrayTests.cs

```diff
@@ -244,34 +145,15 @@ public async Task Value_type_array_nullabilities(ArrayNullabilityMode mode)
         }
     }
 
+    // Note that PG normalizes empty multidimensional arrays to single-dimensional, e.g. ARRAY[[], []]::integer[] returns {}.
     [Test]
-    public async Task Empty_array()
-    {
-        await using var conn = await OpenConnectionAsync();
-        await using var cmd = new NpgsqlCommand("SELECT @p", conn);
-
-        cmd.Parameters.Add(new NpgsqlParameter("p", NpgsqlDbType.Array | NpgsqlDbType.Integer) { Value = Array.Empty<int>() });
-        var reader = await cmd.ExecuteReaderAsync();
-        reader.Read();
-
-        Assert.That(reader.GetFieldValue<int[]>(0), Is.SameAs(Array.Empty<int>()));
-        Assert.That(reader.GetFieldValue<int?[]>(0), Is.SameAs(Array.Empty<int?>()));
-    }
-
-    [Test, Description("Roundtrips an empty multi-dimensional array.")]
-    public async Task Empty_multidimensional_array()
```

Comment:
Can we add this and the zero dimension case to the test cases?

Reply:
This one has been replaced by Write_empty_multidimensional_array, since PG normalizes empty multidimensional arrays to single-dimensional (so we can't use the generic test above with the test cases).

As for the zero dimension case, isn't it the one that's already there (test case Empty_array)?

Reply:
>This one has been replaced by Write_empty_multidimensional_array, since PG normalizes empty multidimensional arrays to single-dimensional (so we can't use the generic test above with the test cases).

👍

>As for the zero dimension case, isn't it the one that's already there (test case Empty_array)?

Yeah Zero_dimensional seems to be a duplicate test.

Reply:
There's no more Zero_dimensional test any more after this PR (I deleted it because the Empty_array case is the same). Is there still a duplicate test after this PR that I'm missing?

Reply:
>Is there still a duplicate test after this PR that I'm missing?
Not at all!